src/lib/core/style-builder/style-builder.ts

```diff
@@ -10,5 +10,14 @@ import {StyleDefinition} from '../style-utils/style-utils';
 
 @Injectable()
 export abstract class StyleBuilder {
-  abstract buildStyles(input: string, parent?: Object): StyleDefinition;
+  abstract buildStyles(input: string, parent?: Object): StyleBuilderOutput;
+
+  sideEffect(_input: string, _styles: StyleDefinition, _parent?: Object) {
+    // This should be a no-op unless an algorithm is provided in a subclass
+  }
+}
+
+export interface StyleBuilderOutput {
+  styles: StyleDefinition;
+  shouldCache: boolean;
```

Comment:
This should be optional with default to `true` and not be required to be returned.